src/shim/crypto.rs

```diff
@@ -209,6 +209,52 @@ pub fn verify_bls_sig(
     fvm_shared_latest::crypto::signature::ops::verify_bls_sig(signature, data, &addr.into())
 }
 
+/// Returns `String` error if a delegated signature is invalid.
+pub fn verify_delegated_sig(
+    signature: &[u8],
+    data: &[u8],
+    addr: &crate::shim::address::Address,
+) -> Result<(), String> {
+    use super::fvm_shared_latest::{
+        address::Protocol::Delegated,
+        crypto::signature::{ops::recover_secp_public_key, SECP_SIG_LEN},
+    };
+    use crate::rpc::eth::types::EthAddress;
+    use crate::utils::encoding::keccak_256;
+
+    if addr.protocol() != Delegated {
+        return Err(format!(
+            "cannot validate a delegated signature against a {} address expected",
+            addr.protocol(),
+        ));
+    }
+
+    if signature.len() != SECP_SIG_LEN {
+        return Err(format!(
+            "invalid delegated signature length. Was {}, must be {}",
+            signature.len(),
+            SECP_SIG_LEN
+        ));
+    }
```

Comment:
nit: there's a nice macro `anyhow::ensure` that makes such validations more expressive.

Reply:
Done.